Compiler/src/translate.ml

```diff
@@ -0,0 +1,1554 @@
+            let temp, inst = proc_expression unwrapped.argument tracker in
+            (true, (temp, inst))
+            
+(* Handle a list of arguments to a function call*)
+and proc_arg_list (arg_list: ('M, 'T) Flow_ast.Expression.ArgList.t) (tracker: Context.tracker) =
+    let (_, unwrapped) = arg_list in
+    let arguments = unwrapped.arguments in
+    let temp_func a = proc_exp_or_spread a tracker in
+    let (is_spread_list, temp) = List.split (List.map temp_func arguments) in
+    let (reg_list, unflattened_inst_list) = List.split temp in
+    (is_spread_list, reg_list, List.flatten unflattened_inst_list)
+
+and proc_exp_call (call_exp: ('M, 'T) Flow_ast.Expression.Call.t) (tracker: Context.tracker) = 
+    let _ : unit = match call_exp.targs with
+        None -> ()
+        | Some a -> raise (Invalid_argument "Unhandled targs in call") in
+    let (_, callee) = call_exp.callee in
+    match callee with
+        (* Handle the method call case explicity*)
+        Flow_ast.Expression.Member member -> 
+            (match member.property with
+                (* Handle method calls seperately for all other cases *)
+                Flow_ast.Expression.Member.PropertyIdentifier (_, id) -> 
+                    let name = id.name in
+                    let arguments = call_exp.arguments in  
+                    let (sub_exp_temp, sub_exp_inst) = proc_expression member._object tracker in
+                    let (is_spread_list, arg_regs, arg_inst) = proc_arg_list arguments tracker in
+                    let is_spread = List.fold_left (||) false is_spread_list in
+                    if is_spread then raise (Invalid_argument "Unhandled spread in member call") else ();
+                    let result_reg = Context.get_new_intermed_temp tracker in
+                    let inouts = [sub_exp_temp] @ arg_regs @ [result_reg] in 
+                    let op = Program_types.Call_method{method_name = name} in
+                    let inst : Program_types.instruction = Program_types.{
+                        inouts =  inouts;
+                        operation = op;
+                    } in
+                    (result_reg, sub_exp_inst @ arg_inst @ [inst])
+                | _ ->
+                    let (callee_reg, callee_inst) = proc_expression call_exp.callee tracker in
+                    let arguments = call_exp.arguments in  
+                    let (is_spread_list, arg_regs, arg_inst) = proc_arg_list arguments tracker in
+                    let is_spread = List.fold_left (||) false is_spread_list in
+                    let result_reg = Context.get_new_intermed_temp tracker in
+                    let inst : Program_types.instruction = if is_spread
+                    then
+                        (let op : Operations_types.call_function_with_spread = Operations_types.{spreads = is_spread_list} in
+                        let inst_op : Program_types.instruction_operation = Program_types.Call_function_with_spread op in
```

Comment:
Probably same here, this would probably be a bit easier if we removed those `WithSpread` variants in FuzzIL for calls as well? Then Method calls and constructors would automatically support them.